### ISSUE TITLE
New posts don't work

### ISSUE THREAD

**Issue by brianrc** [netlify/netlify-cms#171]
**- Do you want to request a *feature* or report a *bug*?**
bug
**- What is the current behavior?**
submitting new post hangs and doesn't upload anything to my git repo
**- If the current behavior is a bug, please provide the steps to reproduce.**
create new post -> enter in test info and submit -> errors in console
**- What is the expected behavior?**
new post uploaded to repo
**- Please mention your node.js, and operating system version.**
site deployed using your defaults: node v6.9.1 and Google Chrome Version 54.0.2840.99 m (64-bit)

![capture](https://cloud.githubusercontent.com/assets/573939/20634396/79eb7e9e-b31e-11e6-9b01-c600210c8fc5.PNG)

**Comment by biilmann** [comment]
Which version of the CMS are you using? This ought to be fixed in the newest release, but let me know if that's not the case...

**Comment by brianrc** [comment]
I guess it's this one: 

`<script src="https://unpkg.com/netlify-cms/dist/cms.js"></script>`

**Comment by cassiozen** [comment]
Hi @brianrc. Can you share your config.yml file? I'll try to replicate your setup here.

**Comment by brianrc** [comment]
Yes, it's right here. https://github.com/brianrc/phenomictest/blob/master/admin/config.yml

**Comment by cassiozen** [comment]
I'm taking a look right now.

**Comment by cassiozen** [comment]
Found the error, @brianrc :

In your config file, you need to add a slug configuration. Something like:

```yaml
collections:
  - name: "post"
    label: "Post"
    folder: "content/posts"
    create: true
    slug: "{{slug}}"
    fields:
      - {label: "Title", name: "title", widget: "string"}
      - {label: "Body", name: "body", widget: "markdown"}
      - {label: "Layout", name: "layout", widget: "string", default: "Post"}
    meta:
      - {label: "Publish Date", name: "date", widget: "datetime"}
```

The slug can contain the special keys day, month and year. For example:

```yaml
slug: "{{year}}-{{month}}-{{day}}-{{slug}}"
```

Now, the CMS can certainly handle this better. I'll make the CMS assume a default slug format if none is provided.

**Comment by brianrc** [comment]
Thank you. That makes sense. It's saving to github now, but it's still not really working though.

It needs to produce output like this:
```
---
title: This is a title
date: 2016-02-14
layout: Post
---

This is the body.
```

Instead it's producing output like so:
```
---
path: content/posts/this-is-a-title.md
title: This is a title
---

This is the body.
```

Where did _path_ come from and why is it ignoring _layout_ and _date_?

**Comment by cassiozen** [comment]
Humm, I see. Let me take a look.

**Comment by cassiozen** [comment]
Hi @brianrc. Just published a new release (0.1.9) and your issues should be resolved.

I'll close this for now. Thanks a lot for working on this with us. If there's any other problems, please reopen.
